src/controller/tests/data_model/TestRead.cpp

```diff
@@ -1617,12 +1615,76 @@ void TestReadInteraction::TestReadHandler_KillOverQuotaSubscriptions(nlTestSuite
         app::InteractionModelEngine::GetInstance()->GetReportingEngine().SetDirty(path);
     }
     readCallback.ClearCounters();
-    ctx.DrainAndServiceIO();
 
-    NL_TEST_ASSERT(apSuite, readCallback.mAttributeCount <= kExpectedParallelPaths);
+    // Run until the global dirtyset is cleared.
```

Comment:
Why does `DrainAndServiceIO` not handle this correctly?  I thought it handled ScheduleWork called from inside something else it processes....

Reply:
DrainAndServiceIO requires "network traffic" between "nodes", if some function called "ScheduleWork" then DrainAndServiceIO will see no traffic and return.